src/main/java/com/commercehub/gradle/plugin/avro/ProcessingState.java

```diff
@@ -24,17 +25,23 @@
 import java.util.Set;
 import java.util.stream.Collectors;
 import org.apache.avro.Schema;
-import org.gradle.api.Project;
+import org.gradle.api.file.ProjectLayout;
 
 class ProcessingState {
     private final Map<String, TypeState> typeStates = new HashMap<>();
     private final Set<FileState> delayedFiles = new LinkedHashSet<>();
     private final Queue<FileState> filesToProcess = new LinkedList<>();
     private int processedTotal;
 
-    ProcessingState(Iterable<File> files, Project project) {
+    ProcessingState(Iterable<File> files, ProjectLayout projectLayout) {
         for (File file : files) {
-            filesToProcess.add(new FileState(file, project.relativePath(file)));
+            final Path filePath;
+            if (file.toPath().isAbsolute()) {
```

Comment:
Is there a problem with running non-absolute paths through `relativize`?

Reply:
Is it possible that a `File` that we were passed could have already been relative, but relative to something other than the project?

Reply:
If we can't leverage `project.relativePath` any more, I'd like to propose extracting the new logic into a new `FileUtils.projectRelativePath(ProjectLayout, File)` method (to encapsulate the logic we land on, so that the intent is clear and easy to read).

Reply:
non-absolute path will throw an exception when you try to relativize them - which is why I added that logic. I will extract it to its own util method 👍